geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSortedSet.java

```diff
@@ -288,6 +289,54 @@ long zcount(SortedSetScoreRangeOptions rangeOptions) {
     return byteIncr;
   }
 
+  long zinterstore(RegionProvider regionProvider, RedisKey key, List<ZKeyWeight> keyWeights,
+      ZAggregator aggregator) {
+    List<RedisSortedSet> sets = new ArrayList<>(keyWeights.size());
+    for (ZKeyWeight keyWeight : keyWeights) {
+      RedisSortedSet set =
+          regionProvider.getTypedRedisData(REDIS_SORTED_SET, keyWeight.getKey(), false);
+
+      if (set == NULL_REDIS_SORTED_SET) {
+        continue;
+      }
+
+      double weight = keyWeight.getWeight();
+      RedisSortedSet weightedSet = new RedisSortedSet(Collections.emptyList());
+
+      for (AbstractOrderedSetEntry entry : set.members.values()) {
+        OrderedSetEntry existingValue = members.get(entry.member);
+        if (existingValue == null) {
+          double score;
+          // Redis math and Java math are different when handling infinity. Specifically:
+          // Java: INFINITY * 0 = NaN
+          // Redis: INFINITY * 0 = 0
+          if (weight == 0) {
+            score = 0;
+          } else if (weight == 1) {
+            score = entry.getScore();
+          } else if (Double.isInfinite(weight) && entry.score == 0D) {
+            score = 0D;
+          } else {
+            double newScore = entry.score * weight;
+            if (Double.isNaN(newScore)) {
+              throw new ArithmeticException(ERROR_OPERATION_PRODUCED_NAN);
```

Comment:
I get that native redis doesn't have the same behavior, but in Java positive infinity plus negative infinity results in NaN, not 0

Reply:
The test that I included here is still failing when run against Geode Redis and passing when run against native Redis. I think that what's needed is a check in the `getSumOfScoresForMember()` method to see if we're about to add positive infinity to negative infinity and if so, not return NaN, but return 0, so that we can match Redis' behaviour. We're already avoiding doing Java math if we know the result would be different from Redis (see the comments just above here about `INFINITY * 0`) so adding in this check isn't deviating from what we've already decided we want to do.

Reply:
I can handle this case specifically, but I don't like it. with the special case of `-infinity + infinity = 0`, this sequence of events `-infinity + infinity + 3` is not equal to `-infinity + 3 + infinity`, which means that the result of our sum will be dependent on the ordering of the sorted sets we're looking through. But that's a problem that native redis has, so we must as well.